### PR TITLE
Exposing fused l2 knn to public APIs

### PR DESCRIPTION
**Pull request by cjnolet**
This is needed for the FAISS support. Since we wrap the fused l2 knn inside our brute-force call, which depends on FAISS currently.

### CODE REVIEW

File: cpp/include/raft/neighbors/brute_force.cuh [hunk 1]
```diff
@@ -191,4 +193,81 @@ void knn(raft::handle_t const& handle,
...
+ * Usage example:
+ * @code{.cpp}
+ *  #include <raft/core/handle.hpp>
+ *  #include <raft/neighbors/brute_force.cuh>
+ *  #include <raft/distance/distance_types.hpp>
+ *  using namespace raft::neighbors;
+ *
+ *  raft::handle_t handle;
+ *  ...
+ *  int k = 10;
```
Comment (divyegala) [inline cpp/include/raft/neighbors/brute_force.cuh:212]: This isn't needed

File: cpp/include/raft/neighbors/brute_force.cuh [hunk 2]
```diff
@@ -191,4 +193,81 @@ void knn(raft::handle_t const& handle,
...
+ */
+template <typename value_t, typename idx_t, typename idx_layout, typename query_layout>
+void fused_l2_knn(const raft::handle_t& handle,
+                  raft::device_matrix_view<const value_t, idx_t, idx_layout> index,
+                  raft::device_matrix_view<const value_t, idx_t, query_layout> query,
+                  raft::device_matrix_view<idx_t, idx_t, row_major> out_inds,
+                  raft::device_matrix_view<value_t, idx_t, row_major> out_dists,
+                  raft::distance::DistanceType metric)
+{
+  int k = out_inds.extent(1);
```
Comment (divyegala) [inline cpp/include/raft/neighbors/brute_force.cuh:236]: Either use auto, idx_t or static_cast if the expectation is this is always an int

File: cpp/include/raft/neighbors/brute_force.cuh [hunk 3]
```diff
@@ -191,4 +193,81 @@ void knn(raft::handle_t const& handle,
...
+               "Distance metric must be L2");
+
+  size_t n_index_rows = index.extent(0);
+  size_t n_query_rows = query.extent(0);
+  size_t D            = index.extent(1);
+
+  RAFT_EXPECTS(raft::is_row_or_column_major(index), "Index must be row or column major layout");
+  RAFT_EXPECTS(raft::is_row_or_column_major(query), "Query must be row or column major layout");
+
+  bool rowMajorIndex = raft::is_row_major(index);
```
Comment (divyegala) [inline cpp/include/raft/neighbors/brute_force.cuh:256]: constexpr

File: cpp/include/raft/neighbors/brute_force.cuh [hunk 4]
```diff
@@ -191,4 +193,81 @@ void knn(raft::handle_t const& handle,
...
+  bool rowMajorQuery = raft::is_row_major(query);
```
Comment (divyegala) [inline cpp/include/raft/neighbors/brute_force.cuh:257]: constexpr

**Review by divyegala — CHANGES_REQUESTED** [review]

**Review by divyegala — APPROVED** [review]

**Comment by cjnolet** [comment]
@gpucibot merge

**Pull request merged**